### PR TITLE
docs: Add task for writing CLC test suite

### PR DESCRIPTION
**Pull request by haighd**
## Summary

Adds a task document tracking the work needed to write a proper test suite for CLC core components.

## Details

Created `docs/TASK-write-test-suite.md` which outlines:
- Current state of tests (23 files, many broken)
- Priority order for new tests (query → hooks → conductor → dashboard)
- Technical approach (pytest, in-memory SQLite, proper fixtures)
- Acceptance criteria

This follows the recommendation from PR #1 to tackle tests in a separate focused effort.

/gemini review

### CODE REVIEW

File: docs/TASK-write-test-suite.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Task: Write Test Suite for CLC Core Components
+
+**Status:** Pending
+**Created:** 2025-12-16
+**Priority:** High
+
+## Summary
+
+The CLC project has 23 test files in `tests/` but many have issues (missing imports, missing fixtures, outdated dependencies). Rather than fix the broken tests, we should write a new focused test suite for core components.
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit 9cdcba5: Added 'Handling Legacy Tests' section specifying that old broken tests should be deleted as they are replaced. Git history preserves them if ever needed.

File: docs/TASK-write-test-suite.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- Use in-memory SQLite for database tests (no external dependencies)
+- Create proper fixtures in `conftest.py`
+- Aim for tests that actually run in CI
+
+## Acceptance Criteria
+
+- [ ] At least 10 passing tests for Priority 1 (query system)
+- [ ] At least 5 passing tests for Priority 2 (hooks)
+- [ ] All tests run successfully in GitHub Actions CI
+- [ ] Tests are documented with clear docstrings
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit 9cdcba5: Replaced test count criteria with coverage targets (80% for query.py, 75% for hooks). Also added edge case coverage requirement.

File: docs/TASK-write-test-suite.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Proposed Approach
+
+Write new, focused tests starting with the most critical components:
+
+### Priority 1: Query System (`query/query.py`)
+- [ ] Test `build_context()` method
+- [ ] Test relevance scoring
+- [ ] Test domain filtering
+- [ ] Test database queries (with SQLite in-memory)
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit ce21e37: Expanded Query System test plan to cover all public methods including query_by_domain, query_by_tags, get_golden_rules, find_similar_failures, validate_database, and CLI functionality.

File: docs/TASK-write-test-suite.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+### Priority 2: Learning Loop Hooks (`hooks/learning-loop/`)
+- [ ] Test `pre_tool_learning.py` - heuristic lookup
+- [ ] Test `post_tool_learning.py` - outcome recording
+- [ ] Test `get_conductor_and_node()` helper
+- [ ] Test trail laying functionality
+
+### Priority 3: Conductor (`conductor/`)
+- [ ] Test workflow start/completion
+- [ ] Test node execution recording
+- [ ] Test run status updates
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit ce21e37: Expanded Conductor test plan to include workflow patterns (linear, branching, conditional, converging), safe_eval_condition helper, and trail management.

File: docs/TASK-write-test-suite.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+- Use `pytest` with `pytest-asyncio` for async tests
+- Use in-memory SQLite for database tests (no external dependencies)
+- Create proper fixtures in `conftest.py`
+- Aim for tests that actually run in CI
+
+## Acceptance Criteria
+
+- [ ] Achieve at least 80% test coverage for `query/query.py` (Priority 1)
+- [ ] Achieve at least 75% test coverage for `hooks/learning-loop/` (Priority 2)
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit 5103eee: Added 70% coverage targets for Priority 3 (Conductor) and Priority 4 (Dashboard API).

File: docs/TASK-write-test-suite.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+- [ ] All new tests run successfully in GitHub Actions CI
+- [ ] New tests are documented with clear docstrings and cover important edge cases
+- [ ] Legacy broken tests are deleted as new tests replace them
+
+## To Start This Task
+
+```bash
+cd ~/.claude/clc
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit 5103eee: Replaced hardcoded cd path with instruction to ensure user is in repo root.

File: docs/TASK-write-test-suite.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+- [ ] Test workflow management (create, get, list)
+- [ ] Test node execution recording (start, completion, failure)
+- [ ] Test run status updates and context management
+- [ ] Test `safe_eval_condition` helper with various inputs
+- [ ] Test different workflow patterns:
+  - [ ] Linear (A -> B -> C)
+  - [ ] Branching (A -> B, A -> C)
+  - [ ] Conditional branching
+  - [ ] Converging (A -> C, B -> C)
+- [ ] Test trail laying and retrieval (`lay_trail`, `get_trails`, `get_hot_spots`)
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed in commit 0c06d24: Split trail laying tests into separate items for lay_trail, get_trails, and get_hot_spots.

File: docs/TASK-write-test-suite.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+**Handling Legacy Tests:** Delete the old, broken test files as they are replaced with new tests. The old tests have unresolvable dependency issues and keeping them around creates confusion about which tests are canonical. Clean deletion is preferred over archiving since the git history preserves the old tests if ever needed.
+
+**Legacy Test Mapping:** Each priority section below includes a list of legacy test files it replaces. Delete these files as the new tests are completed and verified.
+
+**Legacy Test Inventory (23 files):**
+- **To replace** (10 files): Listed in each priority section below
+- **To evaluate** (11 files): `test_baseline_refresh.py`, `test_crash_recovery.py`, `test_dependency_graph.py`, `test_destructive_edge_cases.py`, `test_domain_elasticity.py`, `test_integration_multiagent.py`, `test_lifecycle_adversarial.py`, `test_meta_observer.py`, `test_sqlite_edge_cases.py`, `test_stress.py`, `test_temporal_smoothing.py` - assess during implementation whether these can be salvaged or should be replaced
```
[the remark this answers is not in the document]
Comment (haighd) [reply docs/TASK-write-test-suite.md]: Addressed: Mapped legacy tests to specific priority sections - `test_sqlite_edge_cases.py` and `test_meta_observer.py` to Priority 1, `test_crash_recovery.py` and `test_dependency_graph.py` to Priority 3.
